packages/fetchai/skills/simple_aggregation/handlers.py

```diff
@@ -0,0 +1,301 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the handlers for the oracle aggregation skill."""
+
+from typing import Any, Dict, Optional, cast
+
+from aea.configurations.base import PublicId
+from aea.protocols.base import Message
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.aggregation.message import AggregationMessage
+from packages.fetchai.protocols.default.message import DefaultMessage
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.simple_aggregation.dialogues import (
+    AggregationDialogue,
+    AggregationDialogues,
+    DefaultDialogues,
+    OefSearchDialogue,
+    OefSearchDialogues,
+)
+from packages.fetchai.skills.simple_aggregation.strategy import AggregationStrategy
+
+
+def get_observation_from_message(obs_msg: AggregationMessage) -> Dict[str, Any]:
```

Comment:
There is probably a case for defining `Observation` and `Aggregation` as custom types in the `aggregation` protocol.